vector/src/main/java/im/vector/app/features/widgets/permissions/RoomWidgetPermissionBottomSheet.kt

```diff
@@ -103,6 +103,7 @@ class RoomWidgetPermissionBottomSheet :
 
     override fun onCancel(dialog: DialogInterface) {
         super.onCancel(dialog)
+        doDecline()
```

Comment:
It's weird to call `doDecline()` here, since it will then call `dismiss()` and this is not necessary.

I would had add a call to 

```kotlin
viewModel.handle(RoomWidgetPermissionActions.BlockWidget)
directListener?.invoke(false)
```

instead, or create a new sub function to avoid code duplication.